flow/src/main/java/flow/KeyDispatcher.java

```diff
@@ -65,14 +69,30 @@ private KeyDispatcher(Activity activity, KeyChanger keyChanger) {
     // TODO(#126): this short-circuit may belong in Flow, since every Dispatcher we have implements it.
```

Comment:
If you build the short circuit into flow, then you can also build the rest of this method into flow, and KeyDispatcher/KeyChanger again stop looking necessary.